zio-http/src/main/scala/zio/http/ClientConfig.scala

```diff
@@ -26,6 +27,8 @@ case class ClientConfig(
   def maxThreads(nThreads: Int): ClientConfig = self.copy(nThreads = nThreads)
 
   def useObjectAggregator(objectAggregator: Boolean): ClientConfig = self.copy(useAggregator = objectAggregator)
+
+  def requestDecompression(decompression: Decompression): ClientConfig = self.copy(requestDecompression = decompression)
```

Comment:
I think the Decompression sealed trait is great internally, as it's clearer how to use it as an implementer. However, I think accepting a bool here is fine. As `No` is sort of the default state, and now you're opting into Decompression.

```suggestion
  def requestDecompression(isStrict: Boolean): ClientConfig = {
    val decompression = if (isStrict) Decompression.Strict else Decompression.NonStrict
    self.copy(requestDecompression = decompression)
  }
    
```

Reply:
But other than that, looks great!

Reply:
Changed